client/app/_services/user.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable } from '@angular/core';
+import { Response, URLSearchParams, Headers, RequestOptions }from '@angular/http';
+import { HttpClient } from './http.client';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/observable/of';
+
+import { User } from './userModel';
+@Injectable()
+export class UserService {
+public userData: any;
```

Comment:
According to me this data should either be moved to a data store or should be fetched from the server every time.